core/services/pipeline/runner.go

```diff
@@ -51,6 +51,7 @@ type Runner interface {
 	ExecuteAndInsertFinishedRun(ctx context.Context, spec Spec, vars Vars, l logger.Logger, saveSuccessfulTaskRuns bool) (runID int64, finalResult FinalResult, err error)
 
 	OnRunFinished(func(*Run))
+	InitializePipeline(spec Spec) (*Pipeline, error)
```

Comment:
why do you need to expand this interface for llo? i'm asking b/c i'm imagine a world where plugin development is isolationed from the core node and want to understand what was missing for your use case

Reply:
To cache pipeline and avoid parsing it on every load.